src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -497,36 +520,88 @@ public PointSetTimes eval (
             if (edges[p] == -1) {
                 return true; // Continue to next iteration.
             }
-            edge.seek(edges[p]);
-            int t1 = Integer.MAX_VALUE;
-            int t2 = Integer.MAX_VALUE;
-            // TODO this is not strictly correct when there are turn restrictions onto the edge this is linked to
-            if (distanceTableToVertices.containsKey(edge.getFromVertex())) {
-                t1 = distanceTableToVertices.get(edge.getFromVertex()) + distances0_mm[p] + distancesToEdge_mm[p];
+
+            if (egressArea != null && !GeometryUtils.containsPoint(egressArea, pointSet.getLon(p), pointSet.getLat(p))) {
+                return true; // Point is outside supplied area, continue to next iteration.
             }
-            if (distanceTableToVertices.containsKey(edge.getToVertex())) {
-                t2 = distanceTableToVertices.get(edge.getToVertex()) + distances1_mm[p] + distancesToEdge_mm[p];
+
+            edge.seek(edges[p]);
+
+            int cost = Integer.MAX_VALUE;
+
+            if (routingVariable == RoutingVariable.DISTANCE_MILLIMETERS) {
+                cost = distanceToPoint(costTableToVertices, edge, p);
+            } else if (routingVariable == RoutingVariable.DURATION_SECONDS) {
+                int time0 = costTableToVertices.get(edge.getFromVertex());
+                int time1 = costTableToVertices.get(edge.getFromVertex());
+                if (time0 == 0 && time1 == 0) {
```

Comment:
These zeros as default values seem risky. Is this default value being used anywhere else?

Reply:
I agree the 0 special values are risky. See comment below on setting `Integer.MAX_INTEGER` as the `noEntryValue` in the street router. But as you note, we need to check that changing the `noEntryValue` doesn't have unintended side effects elsewhere.

Reply:
Additional explanation from @ansoncfit: this zero-check is new and was not in pre-existing code. It's an optimization in response to the observation in a comment that `We probably shouldn't evaluate at every point in this LinkedPointSet in case it's much bigger than the driving radius`. Here we still do iterate over every point, but short circuit fast on unreachable points.